basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -234,6 +234,198 @@ namespace basic_autonomy
+
+            }
+
+            //Create Trajectory geometry
+            double delta_step = 1.0 / start_centerline.size();
+
+            for (int i = 0; i < start_centerline.size(); i++)
+            {
+                lanelet::BasicPoint2d current_position;
+                lanelet::BasicPoint2d start_lane_pt = start_centerline[i];
+                lanelet::BasicPoint2d end_lane_pt = end_centerline[i];
+                double delta = delta_step * i;
+                current_position.x() = end_lane_pt.x() * delta + (1 - delta) * start_lane_pt.x();
+                current_position.y() = end_lane_pt.y() * delta + (1 - delta) * start_lane_pt.y();
+
+                centerline_points.push_back(current_position);
+            }
+
+            //Add points from following lanelet to provide sufficient distance for adding buffer
+            auto following_lanelets = wm->getMapRoutingGraph()->following(ending_lanelet, false);
+            if(!following_lanelets.empty()){
+            //Arbitrarily choosing first following lanelet for buffer since points are only being used to fit spline
+            auto following_lanelet_centerline = following_lanelets.front().centerline2d().basicLineString();
+            centerline_points.insert(centerline_points.end(), following_lanelet_centerline.begin(), 
+                                                                        following_lanelet_centerline.end());
+            }
+
+            return centerline_points;
+        }
+
+        std::vector<std::vector<lanelet::BasicPoint2d>> resample_linestrings_to_same_length(std::vector<lanelet::BasicPoint2d> line_1, std::vector<lanelet::BasicPoint2d> line_2){
+            std::vector<std::vector<lanelet::BasicPoint2d>> output;
+            
+            //Fit centerlines to a spline
+            std::unique_ptr<smoothing::SplineI> fit_curve_1 = compute_fit(line_1); // Compute splines based on curve points
+            if (!fit_curve_1)
+            {
+                throw std::invalid_argument("Could not fit a spline curve along the starting_lane centerline points!");
+            }
+
+            std::unique_ptr<smoothing::SplineI> fit_curve_2 = compute_fit(line_2); // Compute splines based on curve points
+            if (!fit_curve_2)
+            {
+                throw std::invalid_argument("Could not fit a spline curve along the ending_lane centerline points!");
+            }
+
+            //Sample spline to get centerlines of equal size
+            std::vector<lanelet::BasicPoint2d> all_sampling_points_line1;
+            std::vector<lanelet::BasicPoint2d> all_sampling_points_line2;
+
+            size_t total_point_size = std::min(line_1.size(), line_2.size());
+
+            all_sampling_points_line1.reserve(1 + total_point_size * 2);
+            std::vector<double> downtracks_raw_line1 = carma_wm::geometry::compute_arc_lengths(line_1);
+            int total_step_along_curve1 = static_cast<int>(downtracks_raw_line1.back() / 2.0);
```

Comment:
Is static_cast what we want here? Or is math::round more appropriate? Static cast here is equivalent to math::floor().